pgmpy/inference/continuous/sampling.py

```diff
@@ -0,0 +1,588 @@
+            self.accepted_proposals += 1.0
+
+        return position, alpha
+
+    def sample(self, initial_pos, num_samples, trajectory_length, stepsize=None, return_type='recarray'):
+        """
+        Method to return samples using Hamiltonian Monte Carlo
+
+        Parameters
+        ----------
+        initial_pos: A 1d array like object
+            Vector representing values of parameter position, the starting
+            state in markov chain.
+
+        num_samples: int
+            Number of samples to be generated
+
+        trajectory_length: int or float
+            Target trajectory length, stepsize * number of steps(L),
+            where L is the number of steps taken per HMC iteration,
+            and stepsize is step size for splitting time method.
+
+        stepsize: float , defaults to None
+            The stepsize for proposing new values of position and momentum in simulate_dynamics
+            If None, then will be choosen suitably
+
+        return_type: string, defaults to 'recarray'
+            A string to get output in different types. Allowed options are
+            'dataframe' for output samples in form of pandas.DataFrame
+            'recarray' for output samples in form of numpy.recarray array
+
+
+        Returns
+        -------
+        list: A list of numpy array type objects containing samples
+
+        Examples
+        --------
+        >>> from pgmpy.inference.continuous import HamiltonianMC as HMC, GradLogPDFGaussian, ModifiedEuler
+        >>> from pgmpy.models import JointGaussianDistribution as JGD
+        >>> import numpy as np
+        >>> mean = np.array([1, -1])
+        >>> covariance = np.array([[1, 0.2], [0.2, 1]])
+        >>> model = JGD(['x', 'y'], mean, covariance)
+        >>> sampler = HMC(model=model, grad_log_pdf=GradLogPDFGaussian, simulate_dynamics=ModifiedEuler)
+        >>> samples = sampler.sample(np.array([1, 1]), num_samples = 5,
+        ...                          trajectory_length=6, stepsize=0.25, return_type='dataframe')
+        >>> samples
+                       x              y
+        0  4.940656e-324  4.940656e-324
+        1   1.592133e+00   1.152911e+00
+        2   1.608700e+00   1.315349e+00
+        3   1.608700e+00   1.315349e+00
+        4   6.843856e-01   6.237043e-01
+        >>> mean = np.array([4, 1, -1])
+        >>> covariance = np.array([[1, 0.7 , 0.8], [0.7, 1, 0.2], [0.8, 0.2, 1]])
+        >>> model = JGD(['x', 'y', 'z'], mean, covariance)
+        >>> sampler = HMC(model=model, grad_log_pdf=GLPG)
+        >>> samples = sampler.sample(np.array([1, 1]), num_samples = 10000,
+        ...                          trajectory_length=6, stepsize=0.25, return_type='dataframe')
+        >>> np.cov(samples.values.T)
+        array([[ 1.00795398,  0.71384233,  0.79802097],
+               [ 0.71384233,  1.00633524,  0.21313767],
+               [ 0.79802097,  0.21313767,  0.98519017]])
+        """
+        return_type = return_type.lower()
```

Comment:
@khalibartan Don't do this. Just try importing pandas and if it fails (handle that exception) and return it as recarray otherwise dataframe.